src/app/services/auth.service.spec.ts

```diff
@@ -1,30 +1,38 @@
 import { TestBed } from '@angular/core/testing';
-import { provideHttpClientTesting } from "@angular/common/http/testing";
+import { HttpTestingController, provideHttpClientTesting } from "@angular/common/http/testing";
 import { AuthService } from './auth.service';
 import { provideHttpClient } from '@angular/common/http';
+import { RegisterRequest } from '../register-request';
+
 
 
 describe('AuthService', () => {
   let service: AuthService;
-  let httpClientSpy: any
+  let  httpTestController: HttpTestingController
 
-  beforeEach(() => {
+  beforeEach(() => {  
     TestBed.configureTestingModule({
       imports: [],
       providers: [AuthService, provideHttpClient(),  provideHttpClientTesting()]
     });
     service = TestBed.inject(AuthService);
+    httpTestController = TestBed.inject(HttpTestingController)
   });
 
-  beforeEach(() => {
-    httpClientSpy = {
-      post: jest.fn()
-    }
-  })
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
 
- 
+   it('should make a POST  request to register endpoint when valid email and password are provided',() => {
+     const command: RegisterRequest = {
+        email: 'michael.lawson@reqres.in',
+        password: ' 4ahr8r;'
+     }
+
+     service.register(command).subscribe()
```

Comment:
Will it not be nice to test the response as well?

Reply:
@Bilal-Abubakari response has been added to the new PR